src/utils/prices.ts

```diff
@@ -34,49 +30,31 @@ export function getPriceImpactWarning(priceImpact?: Percent): 'warning' | 'error
 }
 
 // computes realized lp fee as a percent
-export function computeRealizedLPFeePercent(trade: Trade<Currency, Currency, TradeType>): Percent {
+export function computeRealizedLPFeePercent(trade: InterfaceTrade): Percent {
   let percent: Percent
 
-  // Since routes are either all v2 or all v3 right now, calculate separately
-  if (trade.swaps[0].route.pools instanceof Pair) {
-    // for each hop in our trade, take away the x*y=k price impact from 0.3% fees
-    // e.g. for 3 tokens/2 hops: 1 - ((1 - .03) * (1-.03))
-    percent = ONE_HUNDRED_PERCENT.subtract(
-      trade.swaps.reduce<Percent>(
-        (currentFee: Percent): Percent => currentFee.multiply(INPUT_FRACTION_AFTER_FEE),
-        ONE_HUNDRED_PERCENT
-      )
-    )
-  } else {
-    percent = ZERO_PERCENT
-    for (const swap of trade.swaps) {
-      const { numerator, denominator } = swap.inputAmount.divide(trade.inputAmount)
-      const overallPercent = new Percent(numerator, denominator)
+  percent = ZERO_PERCENT
+  for (const swap of trade.swaps) {
+    const { numerator, denominator } = swap.inputAmount.divide(trade.inputAmount)
+    const overallPercent = new Percent(numerator, denominator)
 
-      const routeRealizedLPFeePercent = overallPercent.multiply(
-        ONE_HUNDRED_PERCENT.subtract(
-          swap.route.pools.reduce<Percent>((currentFee: Percent, pool): Percent => {
-            const fee =
-              pool instanceof Pair
-                ? // not currently possible given protocol check above, but not fatal
-                  FeeAmount.MEDIUM
-                : pool.fee
-            return currentFee.multiply(ONE_HUNDRED_PERCENT.subtract(new Fraction(fee, 1_000_000)))
-          }, ONE_HUNDRED_PERCENT)
-        )
+    const routeRealizedLPFeePercent = overallPercent.multiply(
+      ONE_HUNDRED_PERCENT.subtract(
+        swap.route.pools.reduce<Percent>((currentFee: Percent, pool): Percent => {
+          const fee = pool instanceof Pair ? /* V2 FeeAmount= */ FeeAmount.MEDIUM : pool.fee
```

Comment:
can you make ` pool instanceof Pair `a variable called `isV2`? then explain why that pool won't have a fee attribute

Reply:
ideally also name `FeeAmount` -> `V2FeeAmount`

Reply:
I'm not sure, because moving the check to a variable may lose the narrowing of the type to `Pair`. Made a new helper method `getFeeAmount` to get around this.

Reply:
works for me